app/js/enhancedDate.js

```diff
@@ -1,5 +1,79 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+//  Brian Forcum Module 2
+ 
 
+var enhancedDate = (function iffe() {
+  
+  'use strict';
+  
+  var theDate = {
+    time: new Date(),
+    days: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 
+           'Friday', 'Saturday'],
+    months: ['January', 'February', 'March', 'April', 'May', 'June',
+             'July', 'August', 'September', 'October', 'November',
+             'December'],
+    
+  };
+  
+ 
+  var publicAPI = {
+    setDate: function setDate(d) {
+      
+      if (!(d instanceof Date)) {
+          theDate.time = new Date();
+      } else {
+        theDate.time = d;
```

Comment:
This works, but keep in mind that objects are passed as arguments by reference, so if the original Date instance passed in is modified externally, it will modify your internal date instance as well. Instead, just create a new instance, passing in that value `theDate.time = newDate(d);`  Also, it should allow milliseconds to be used to instantiate a new Date as well.